### ISSUE TITLE
Webpack build is silent when cache.type = 'filesystem'

### ISSUE THREAD

**Issue by DimaGashko** [webdiscus/html-bundler-webpack-plugin#115]
## Current behaviour

I know it's experimental, but I just not sure if it's expected behavior for now or not.
When I set `cache.type = 'filesystem'` the build doesn't have any logs in the terminal.

Here I run the build a few times - with cache.type=memory, cache.type=filesystem, and one more time after deleting the cache directory. Only first time I get any output in the terminal. Each time I got the correct build.

![image](https://github.com/user-attachments/assets/3c7043e3-bd3d-4955-862e-ef50f0746665)

If the build fails (e.g. change the img path to non-existing), I still see no errors, but when I fix the issue  and try to rebuild (or just rebuild instantly) - it fails.

When I delete the plugin and uncomment the entry property of the config - everything is working as expected.

## Expected behaviour

I expect to see the same output in the terminal as usual, and also I expect the build to fail when there was problems during the build.

### Reproduction

Config I used for testing:

```js
const HtmlBundlerPlugin = require('html-bundler-webpack-plugin');

/** @type {import('webpack').Configuration} */
module.exports = {
   mode: 'production',
   // entry: './src/index.js',
   output: {
      path: `${__dirname}/dist`,
      clean: true,
   },
   stats: {
      preset: 'normal',
      children: true,
   },
   resolve: {
      extensions: ['.js'],
      alias: {
         '@': `${__dirname}/src`,
      },
   },
   cache: {
      // type: 'memory',
      type: 'filesystem',
      buildDependencies: {
         config: [__filename],
      },
   },
   module: {
      rules: [{
         test: /\.(png|jpe?g|webp|avif|svg|gif|ico)$/i,
         type: 'asset/resource',
      }],
   },
   plugins: [
      new HtmlBundlerPlugin({
         entry: {
            'index': './src/index.html',
         },
      }),
   ],
};

```

## Environment

- OS: [macOS, Linux, Windows]
Linux (ArchLinux)

## Additional context

p.s. what also your predictions/plans for cache.type=filesystem? Will it be fully supported sometime in the future? In a year, two, n years? So we can understand what to expect.

**Comment by webdiscus** [comment]
Hello @DimaGashko,

thanks for the issue report.
Can you please create a small repo with reproducible issue?

### P.S.
The supports of `filesystem` caching is experimental because I cannot guarantee 100% functionality in all cases of use.
The `filesystem` caching is very very complex to integrate with this powerful plugin on all levels.
Thanks to the issues reported by users, I can implement missing or incorrect functionality.
Yes of cause I will release the  supports of `filesystem` caching, it is very important feature, but I need help of community to test it and report issues.

Feel you free to create feature requests or a bug issues.

**Comment by DimaGashko** [comment]
Here's the repo:
https://github.com/DimaGashko/html-bundler-plugin.filesystem

![image](https://github.com/user-attachments/assets/eee69049-0748-48f2-9e61-eb8063ab2522)

**Comment by webdiscus** [comment]
@DimaGashko 

yes, I can reproduce the issue. I will research the "silent mode" and try to fix it.

Using `memory` caching:
![image](https://github.com/user-attachments/assets/81f80e55-4b03-4b61-aa3b-5e01cbcddc6b)

Using `filesystem` caching:
![image](https://github.com/user-attachments/assets/5d8ffb95-f743-406d-82a8-6b5eaf9cd8f9)

**Comment by webdiscus** [comment]
@DimaGashko 

currently you can use the `verbose: true` plugin option to display all precessed assets:
```js
new HtmlBundlerPlugin({
      entry: {
        index: './src/index.html',
      },
      verbose: true,
    }),
```

The plugin has own verbose to display all entry points and dependencies.

**Comment by webdiscus** [comment]
@DimaGashko  the issue is fixed in the `v4.1.3`

**Comment by DimaGashko** [comment]
@webdiscus looks great. The output as displayed as expected. When the build fails as well. And even more - the cache was broken after failed build but now it works properly as well.

Thanks for that fast response and even fix!
